subprojects/core/src/main/java/org/gradle/internal/classpath/Instrumented.java

```diff
@@ -116,6 +129,52 @@ public static void groovyCallSites(CallSiteArray array) {
         }
     }
 
+    /**
+     * The bootstrap method for method calls from Groovy compiled code with indy enabled.
+     * Gradle's bytecode processor replaces the Groovy's original {@link IndyInterface#bootstrap(MethodHandles.Lookup, String, MethodType, String, int)}
+     * with this method to intercept potentially "interesting" calls and do some additional work.
+     *
+     * @param caller the lookup for the caller (JVM-supplied)
+     * @param callType the type of the call (corresponds to {@link IndyInterface.CallType} constant)
+     * @param type the call site type
+     * @param name the real method name
+     * @param flags call flags
+     * @return the produced CallSite
+     * @see IndyInterface
+     */
+    public static java.lang.invoke.CallSite bootstrap(MethodHandles.Lookup caller, String callType, MethodType type, String name, int flags) {
+        CacheableCallSite cs = toGroovyCacheableCallSite(IndyInterface.bootstrap(caller, callType, type, name, flags));
+        switch (callType) {
+            case "invoke":
```

Comment:
What about the `setProperty` call type? For example, is it possible to encode a call to `System.setProperties` as an indy `setProperty`?